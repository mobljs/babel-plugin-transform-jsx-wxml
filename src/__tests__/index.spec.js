const babel = require('@babel/core');
const plugin = require('../');

const source = `
const array = ['1', '2', '3'];
<View>
  {state.flag && <MyButton>Click Me</MyButton>}
  
  {['a', 'b', 'c', 1, 2].map((it, idx) => (<Body1>{it}</Body1>))}
  
  {array.map((it, idx) => <Image lazyLoad>{it}</Image>)}      
</View>
`;

it('should tern jsx to wxml', () => {
  const { code } = babel.transform(source, {
    plugins: [plugin],
  });
  expect(code).toMatchSnapshot();
});
