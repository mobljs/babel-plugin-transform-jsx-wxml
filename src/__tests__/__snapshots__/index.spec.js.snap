// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should tern jsx to wxml 1`] = `
"const array = ['1', '2', '3'];
<view>
  <my-button wx:if=\\"{{state.flag}}\\">Click Me</my-button>
  
  <body1 wx:for=\\"{{['a','b','c',1,2]}}\\" wx:for-item=\\"{{it}}\\" wx:for-index=\\"{{idx}}\\">{{it}}</body1>
  
  <image lazy-load wx:for=\\"{{array}}\\" wx:for-item=\\"{{it}}\\" wx:for-index=\\"{{idx}}\\">{{it}}</image>      
</view>;"
`;
